packages/app/src/utils.ts

```diff
@@ -720,3 +720,11 @@ export function getMetricTableName(
 export function toArray<T>(obj?: T | T[]): T[] {
   return !obj ? [] : Array.isArray(obj) ? obj : [obj];
 }
+
+// Helper function to remove trailing slash
+export const stripTrailingSlash = (url: string | undefined | null): string => {
+  if (!url) {
+    return '';
+  }
```

Comment:
maybe we shouldn't mutate the return value if its nullish. I'd imagine the caller should throw if host is invalid?

Reply:
agree. this should throw instead of a valid type that will just fail later on.